src/daemon/rpmostreed-utils.c

```diff
@@ -202,16 +202,6 @@ rpmostreed_refspec_parse_partial (const gchar *new_provided_refspec,
         }
     }
 
-  if (g_strcmp0 (origin_remote, remote) == 0 &&
-      g_strcmp0 (origin_ref, ref) == 0)
```

Comment:
Hmm, I'm not sure why that check was there also. Seems like the only other use is in the legacy API `os_handle_get_cached_rebase_rpm_diff()`.